src/server.ts

```diff
@@ -404,10 +404,10 @@ connection.onDidChangeConfiguration(change => {
 
     specificValidatorPaths = [];
     if (settings.yaml) {
-        yamlConfigurationSettings = settings.yaml.schemas;
-        yamlShouldValidate = settings.yaml.validate;
-        yamlShouldHover = settings.yaml.hover;
-        yamlShouldCompletion = settings.yaml.completion;
```

Comment:
These were overwritten if, e.g., `yaml.validate.enabled` was not specified in the configuration message.